xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -671,11 +671,10 @@ bool IsBmm2GradGemm2(HloInstruction* instr) {
 }
 
 MatchBwdResult MatchBmm1GradGemm1(MatchBwdResult previous_result,
-                                  HloInstruction* fwd_fmha_call,
                                   HloInstruction* bmm_1) {
   MatchBwdResult match_result = previous_result;
   match_result.has_match = false;
-  const HloInstruction* q_tensor = fwd_fmha_call->operand(0);
+  const HloInstruction* q_tensor = bmm_1->operand(0);
```

Comment:
Another test that covers this change would be nice :)

Reply:
Add one test case where Q is transposed because head dim does not have stride 1(layout constraint required by cuDNN backend). Improved bwd pattern match will pattern match in this case.